app/models/user.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable, :confirmable
+  include DeviseTokenAuth::Concerns::User
+
+  enum gender: %i[male female]
+
+  validates :name, :gender, presence: true
+  validates :name, length: { maximum: 30 }
+  validates :gender, inclusion: { in: genders.keys }, if: :gender_present?
```

Comment:
You can use`if: Proc.new{ |u| u.gender.present? } `. That way you won't need to define a new method